cli/src/commands/backout.rs

```diff
@@ -47,12 +47,37 @@ pub(crate) fn cmd_backout(
         parents.push(destination);
     }
     let mut tx = workspace_command.start_transaction();
-    back_out_commit(
-        command.settings(),
-        tx.mut_repo(),
-        &commit_to_back_out,
-        &parents,
-    )?;
+    let commit_to_back_out_subject = commit_to_back_out
+        .description()
+        .lines()
+        .next()
+        .unwrap_or_default();
+    let is_reapplying_backed_out_commit = commit_to_back_out_subject.starts_with("Back out \"")
+        && commit_to_back_out_subject.ends_with('"');
+    let new_commit_subject = if is_reapplying_backed_out_commit {
+        &commit_to_back_out_subject["Back out \"".len()..commit_to_back_out_subject.len() - 1]
```

Comment:
nit: I don't know if we need this hack, but `strip_prefix`/`suffix` can be used instead.

Reply:
I'll leave this open for a bit more in case anyone feels that the differentiated "Back out"/"Reapply" subject headers are unnecessary.

Reply:
fwiw, we'll also need to parse `Reapply "` so triple backouts just says `Backout "`?

Reply:
Git doesn't parse it (https://github.com/git/git/compare/86b56ff267962ab91aa368b10d6df10b1e9e291e...c9192f9e45f12f7e4c93488eb77a3098e75a78cc), so I didn't attempt to parse it as well. I think the current single layer of unwrapping is a decent default.

Reply:
Hmm, I have no idea why reverting a revert is special. `"Revert "Reapply ..""` looks as ugly as `"Revert "Revert ..""`. Anyway, I wouldn't add this feature because it won't work if the user edited a commit message, and ideally they should.

Reply:
Unless anyone has strongly feels otherwise then, I can remove the "Reapply" logic and merge this in tomorrow. `jj` does make very easy to edit commit messages, so perhaps we don't need the special-casing. The lengthy "Backout ..." message might also encourage people to edit the message to better describe the commit. We can always revisit this decision if necessary.